src/lib/storage.test.ts

```diff
@@ -21,6 +21,15 @@ describe("Storage", () => {
     expect(mockSender).toHaveBeenCalledWith("add", ["key1"], "value1");
   });
 
+  it("should set an item and wait", () => {
+    const mockSender = jest.fn();
+    storage.registerSender(mockSender);
+
+    storage.setItemAndWait("key1", "value1");
```

Comment:
```suggestion
  it("should set an item and wait", async () => {
    const mockSender = jest.fn();
    storage.registerSender(mockSender);

    await storage.setItemAndWait("key1", "value1");
```

I think... you have to await the result of this call else the set call & the expect assertion are going to be racing each other.

Reply:
i saw this, the tests oddly fail using this condition because of how they are mocked. I originally had this implementation.

Reply:
Ahh, yeah... I mean, we might need to update how the mock happens then, I suppose.  It's too weird to have a test that doesn't really simulate the real thing. :see_no_evil: